src/main/scala/org/constellation/infrastructure/cluster/ClusterStorageInterpreter.scala

```diff
@@ -1,18 +1,46 @@
 package org.constellation.infrastructure.cluster
 
+import cats.data.NonEmptyList
 import cats.effect.concurrent.Ref
 import cats.effect.{Clock, Sync}
 import cats.syntax.all._
 import org.constellation.domain.cluster.ClusterStorageAlgebra
-import org.constellation.p2p.{JoinedHeight, PeerData}
+import org.constellation.p2p.Cluster.MissingActivePeers
+import org.constellation.p2p.{JoinedHeight, MajorityHeight, PeerData}
+import org.constellation.schema.NodeType.{Full, Light}
+import org.constellation.schema.snapshot.NextActiveNodes
 import org.constellation.schema.{Id, NodeState, NodeType}
+import org.constellation.util.Metrics
 
 import scala.concurrent.duration._
 
-class ClusterStorageInterpreter[F[_]]()(implicit F: Sync[F], C: Clock[F]) extends ClusterStorageAlgebra[F] {
+class ClusterStorageInterpreter[F[_]](nodeId: Id, nodeType: NodeType)(implicit F: Sync[F], C: Clock[F])
+    extends ClusterStorageAlgebra[F] {
 
   private val peers: Ref[F, Map[Id, PeerData]] = Ref.unsafe(Map.empty[Id, PeerData])
 
+  // TODO: move initialization of the refs below to somewhere outside - Cluster class startup?
```

Comment:
why?

Reply:
It's in startup flow currently. It was initialized during Cluster class startup till refs where moved to storage classes and because clusterStorage and nodeStorage are only aware of themselves the initial state (like initial full nodes set based on whitelisting file) can't be set during storage initial creation. I did put keyPair or nodeId in some storages but I'm not convinced it should be like that, so I still plan to move these outside of storage and handle logic that needs these parameters differently.